build/teamcity/cockroach/nightlies/compose.sh

```diff
@@ -6,22 +6,17 @@ dir="$(dirname $(dirname $(dirname $(dirname "${0}"))))"
 source "$dir/teamcity-support.sh"
 source "$dir/teamcity-bazel-support.sh"
 
-# The test failures generated by TestComposeCompare are not necessarily
-# failures per se. They're cases of behavioral divergences from Postgres. While
-# our compatibility guarantees are not 100%, it's better to treat failures as
-# information to occasionally review.
-export SKIP_LABEL_TEST_FAILURE=1
-
 tc_start_block "Run compose tests"
 
 bazel build //pkg/cmd/bazci --config=ci
 BAZEL_BIN=$(bazel info bazel-bin --config=ci)
 BAZCI=$BAZEL_BIN/pkg/cmd/bazci/bazci_/bazci
 
-bazel build //pkg/cmd/cockroach //pkg/compose/compare/compare:compare_test --config=ci --config=crosslinux --config=test
+bazel build //pkg/cmd/cockroach //pkg/compose/compare/compare:compare_test //c-deps:libgeos //build/bazelutil:test_force_build_cdeps --config=ci --config=crosslinux --config=test -c opt
```

Comment:
There's no reason to build `//build/bazelutil:test_force_build_cdeps`.

I also don't see a reason to add the `-c opt`, you can just omit that.

Reply:
I added the `//build/bazelutil:test_force_build_cdeps` since that's what `dev` seems to do here: https://github.com/cockroachdb/cockroach/blob/678be4e74c804481e127846525179949a9759e97/pkg/cmd/dev/build.go#L420-L423

so anyone using `dev compose` would get that behavior

Reply:
It doesn't matter. When you cross-build, the build runs in a Docker container and any settings you have to `--force_build_cdeps` in `.bazelrc.user` won't be applied. That means that the default (which is to not pass `--force_build_cdeps`) will happen.

If it *did* matter, you would have to look in the output file and check the output to see if the flag were set, as in [here](https://github.com/cockroachdb/cockroach/blob/678be4e74c804481e127846525179949a9759e97/pkg/cmd/dev/util.go#L148). Just building it without checking the output is not a meaningful/useful thing to do.